spark-3.2/clickhouse-spark/src/main/scala/xenon/clickhouse/read/InputPartitions.scala

```diff
@@ -40,6 +40,11 @@ case class ClickHouseInputPartition(
 
   def partFilterExpr: String = partition match {
     case NoPartitionSpec => "1=1"
-    case PartitionSpec(part, _, _) => s"${table.partition_key} = $part"
+    case PartitionSpec(part, _, _) => (part.contains("-"), part.contains("(")) match {
+      // quote when partition by a single Date Type column to avoid illegal types of arguments (Date, Int64)
```

Comment:
It's enough for the `Date` case, and we can imporve it in future by extract a method like `compilePartValue`

Reply:
clickhouse 21.6 above add virtual column `_partition_id`, perhaps more suitable for `InputPartitions` ?

Reply:
We need to know the partition value for pruning

Reply:
After second thought, it should be a good idea, because that partition pruning ocuurs before inferring `InputPartition`

Reply:
Compare to the current implementation, switching to `_partition_id` does not have side-effect except for compatibility, we can do that and introduce a configure in case the user uses v21.3

Reply:
`planInputPartitionsById` and default to `true` ?

Reply:
SGTM.